p2p/pkg/autodepositor/autodepositor.go

```diff
@@ -201,28 +240,9 @@ func (adt *AutoDepositTracker) startAutodeposit(egCtx context.Context, eg *errgr
 				return fmt.Errorf("error in autodeposit event subscription: %w", err)
 			case window := <-adt.windowChan:
 				adt.currentOracleWindow.Store(window.Window)
-				withdrawWindows, err := adt.store.ListDeposits(egCtx, new(big.Int).Sub(window.Window, big.NewInt(1)))
-				switch {
-				case err != nil:
-					adt.logger.Error("failed to list deposits", "err", err)
-					return err
-				case len(withdrawWindows) == 0:
-					adt.logger.Info("no deposits to withdraw")
-				case len(withdrawWindows) > 0:
-					adt.logger.Info("deposits to withdraw", "windows", withdrawWindows)
-					opts, err := adt.optsGetter(egCtx)
-					if err != nil {
-						return err
-					}
-					txn, err := adt.brContract.WithdrawFromWindows(opts, withdrawWindows)
-					if err != nil {
-						return err
-					}
-					adt.logger.Info("withdraw from windows", "hash", txn.Hash(), "windows", withdrawWindows)
-					err = adt.store.ClearDeposits(egCtx, withdrawWindows)
-					if err != nil {
-						return fmt.Errorf("failed to clear deposits: %w", err)
-					}
+				err := adt.withdrawFromPastWindows(egCtx, window.Window)
```

Comment:
Isn't `withdrawFromPastWindows` already called in `Start`? What is the scenario where `withdrawFromPastWindows` should be called twice?

Reply:
Autodeposit is an ongoing process, and once a window has passed, we want to withdraw the funds from that window

Reply:
Ah my bad makes sense, I thought existing functionality lived elsewhere. All looks good